### ISSUE TITLE
Snapweb was not installed

### ISSUE THREAD

**Issue by audas** [badaix/snapcast#1164]
Snap web was not installed

I installed Snapserver on Rasperry OS on a 4B+ and there was no Snapweb with it.

Not sure if this is "meant" to be included as it appears it is in the documentation or if we are meant to make and include it.

**Comment by badaix** [comment]
@audas I think the "new issue template" should have asked you for the Snapcast version and other details. If you installed some older version from your distribution, it might be that there is no snapweb included.

**Comment by audas** [comment]
Sorry about the versioning.

I am using version v0.26.0 - I was following along with the docs on a raspberry pi and could not get the version to work so just went for snapclient and snapserver. 

I installed node-typetext and used the make file, but could not get the config file to recognise the address and so must use a longer path. But it does not really work.

http://<ip>:1780/snapweb/dist/index.html

I am really struggling with this. I tried BalenaSound and it took over the devices too much and wanted to try this. Have been working for 12 hours and can not get any sound.

**Comment by audas** [comment]
> @audas I think the "new issue template" should have asked you for the Snapcast version and other details. If you installed some older version from your distribution, it might be that there is no snapweb included.

It looks like there is a post below mine, which references this issue - we can not install the packages by version on Raspberry with apt-get. 

This means we are getting an old version.

**Comment by audas** [comment]
apt is out of sync

```
$ sudo apt install snapserver
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
snapserver is already the newest version **(0.26.0+dfsg1-1+b1).**
```

**Comment by audas** [comment]
> @audas I think the "new issue template" should have asked you for the Snapcast version and other details. If you installed some older version from your distribution, it might be that there is no snapweb included.


So I just tried the build from source and there are definitely two versions.
So we are unable to install the latest version using apt-get as the file is not listed in the debian repository (not git),
if we build from source from github it gives us the latest version .27 - so we have to build everything from source to get the latest version.

**Comment by kingosticks** [comment]
Debian is a stable distribution, not a cutting edge distribution. This is the point of Debian.

The only way you will get the latest version from Debian's own repos is if the software stops being worked on.

**Comment by audas** [comment]
> Debian is a stable distribution, not a cutting edge distribution. This is the point of Debian.
> 
> The only way you will get the latest version from Debian's own repos is if the software stops being worked on.
> 
> Some projects provide pre-built binaries and even Debian packages, but these won't be available from Debian's apt repos, you'll need to go find them and update them manually.


I am referencing the ones linked to and provided by the readme.

Link 
https://github.com/badaix/snapcast/blob/develop/doc/install.md#debian

Says do this

sudo apt install </path/to/snapclient_0.x.x_[arch].deb>

But they dont work.

https://github.com/badaix/snapcast/releases/tag/v0.27.0

As per other thread 

https://github.com/badaix/snapcast/issues/1163

**Comment by kingosticks** [comment]
Sure, I was just trying to explain why

> $ sudo apt install snapserver

Has no chance of giving you what you want.

**Comment by jreusch** [comment]
> apt is out of sync
> 
> ```
> $ sudo apt install snapserver
> Reading package lists... Done
> Building dependency tree... Done
> Reading state information... Done
> snapserver is already the newest version **(0.26.0+dfsg1-1+b1).**
> ```

you're installing die version packaged by debian (dfsg)
you can see here which are available where:
https://packages.debian.org/search?keywords=snapserver

after looking into those packages it seems they do not include snapweb
>   * Repackage orig tarball to exclude libflac.js which is missing the source code.
>    - Disable snapweb as it relies on the javascript library.

https://metadata.ftp-master.debian.org/changelogs//main/s/snapcast/snapcast_0.26.0+dfsg1-1_changelog
so either install snapserver version here from github to get snapweb, extract snapweb from here, place it somewhere and adapt path, or maybe try to get debian package fixed :)

**Comment by badaix** [comment]
Starting with Snapweb version 0.6.0, there are debian packages as well as raw archives of Snapweb available:
https://github.com/badaix/snapweb/releases

The debian package can be installed with `sudo apt install ./snapweb_0.6.0-1_all.deb`. It will be installed to `/usr/share/snapweb`, so you have to change the `doc_root` in `/etc/snapserver.conf` accordingly, to point to this directory.

**Comment by DesertRider** [comment]
I had the same problem under Bullseye (Raspian 11) on a Raspberry Pi 3 and couldn't install package snapweb_0.6.0-1_all.deb because of this error:
dpkg-deb: error: archive '/opt/snapweb_0.6.0-1_all.deb' uses unknown compression for member 'control.tar.zst'

As I couldn't find a solution concerning the compression problem and dpkg, my solution was to manually download snapweb.zip and decompress it in /usr/share/snapweb/. Pointing this directory in /etc/snapserver.conf made it works.

**Comment by badaix** [comment]
@DesertRider Can you please open an issue for this in [Snapweb](https://github.com/badaix/snapweb/issues)?

**Comment by DesertRider** [comment]
> @DesertRider Can you please open an issue for this in [Snapweb](https://github.com/badaix/snapweb/issues)?

Done!

**Comment by badaix** [comment]
in v0.27.0 I've copied the Snapweb bundle from the Snapweb repo into the source tree, and there I've used some downloaded version of Flac.js, which seems to be the reason that it's not bundle in the official Debian packages.
I changed in v0.28.0 two things: Snapweb uses now Flac.js from the NPM repo and the Snapweb distribution is downloaded and added to the debian package in the `Package.yml` github action. If you build the package without adding Snapweb, a default page will be shown with a how-to install Snapweb guide.
